src/jit/codegenlegacy.cpp

```diff
@@ -19041,7 +19041,10 @@ regMaskTP CodeGen::genCodeForCall(GenTreeCall* call, bool valUsed)
                             indCallReg = regSet.rsGrabReg(RBM_ALLINT); // Grab an available register to use for the CALL
                                                                        // indirection
                             instGen_Set_Reg_To_Imm(EA_HANDLE_CNS_RELOC, indCallReg, (ssize_t)addr);
-                            getEmitter()->emitIns_R_R_I(INS_ldr, EA_PTRSIZE, indCallReg, indCallReg, 0);
+#ifdef _TARGET_ARM_
+                            if (!compiler->opts.ShouldUsePInvokeHelpers())
```

Comment:
This does not look right. Why do you want to skip the indirection for PInvokeHelpers?